sympy/polys/rings.py

```diff
@@ -846,6 +853,11 @@ def __sub__(
         if self.ring.is_element(other):
             return self._sub(other)
 
+        if isinstance(other, PolyElement):
+            domain = other.ring.domain
+            if isinstance(domain, PolynomialRing) and domain.ring.is_element(self):
+                return - other._sub_ground(self) # type: ignore[arg-type]
```

Comment:
This should call `other._rsubground(self)` rather than `- other._sub_ground(self)`. We don't want to create an unnecessary intermediate polynomial.